hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleTokenRelStatus.java

```diff
@@ -182,7 +194,10 @@ public void setAutomaticAssociation(boolean automaticAssociation) {
 	@Override
 	public MerkleTokenRelStatus copy() {
 		setImmutable(true);
-		return new MerkleTokenRelStatus(balance, frozen, kycGranted, automaticAssociation, numbers);
+		final var copy = new MerkleTokenRelStatus(balance, frozen, kycGranted, automaticAssociation, numbers);
```

Comment:
Would it make more sense to define a protected copy constructor, i.e. 

```
  protected MerkleTokenRelStatus(MerkelTokenRelStatus obj)
```

And then use that to copy? i.e.

```
public MerkleTokenRelStatus copy() {
    setImmutable(true);
    MerkleTokenRelStatus copy = new MerkleTokenRelStatus(this);
    copy.setImmutable(false);
    return copy;
}
```
This will be cleaner and easier to understand,.